src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -305,23 +305,32 @@ void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion
         break;
     }
 
+    // 2.B) Clamp the movement to the mutable viewport
+    const auto bufferSize = _activeBuffer().GetSize();
+    const auto mutableViewport = _GetMutableViewport();
+    const COORD bottomRightInclusive{ mutableViewport.RightInclusive(), mutableViewport.BottomInclusive() };
+    if (bufferSize.CompareInBounds(targetPos, bottomRightInclusive) > 0)
+    {
+        targetPos.Y = bottomRightInclusive;
+    }

```

Comment:
BTW now that I'm seeing this, if you make such a change in `main` you can just do
```cpp
targetPos = std::min(targetPos, bottomRightInclusive);
```

Reply:
I'll move this comment as a code suggestion over to #13318

Reply:
I messed up. Meant to remove the `.Y`. Just did that in the latest commit